stdlib/2/itertools.pyi

```diff
@@ -71,7 +71,12 @@ def izip(iter1: Iterable[_T1], iter2: Iterable[_T2],
 @overload
 def izip(iter1: Iterable[_T1], iter2: Iterable[_T2], iter3: Iterable[_T3],
          iter4: Iterable[_T4]) -> Iterator[Tuple[_T1, _T2,
-                                           _T3, _T4]]: ...  # TODO more than four iterables
+                                           _T3, _T4]]: ...
+@overload
+def izip(iter1: Iterable[_T1], iter2: Iterable[_T2], iter3: Iterable[_T3],
+         iter4: Iterable[_T4], iter5: Iterable[_T5])
+         -> Iterator[Tuple[_T1, _T2, _T3, _T4, _T5]]: ...  # TODO more than five iterables
```

Comment:
I believe that while that would work for n-ary calls, it would drop type information for the individual tuple positions, so spelling it out may be worth the tradeoff here and work in more cases.  The use cases I've seen for izip are often of the form

    for foo, bar in itertools.izip(foos, bars):
      frobnicate(foo, bar)

and it would make sense to keep the type information for foo and bar around.